src/util/v4/v4_altivec.h

```diff
@@ -1,166 +1,149 @@
+    friend inline v4 splat( const v4 &a ) ALWAYS_INLINE;
 
-/*     ASSIGN(=, v = b.v ); */
+    template<int i0, int i1, int i2, int i3>
+    friend inline v4 shuffle( const v4 &a ) ALWAYS_INLINE;
 
-/* #   undef ASSIGN */
+    friend inline void swap( v4 &a, v4 &b ) ALWAYS_INLINE;
+    friend inline void transpose( v4 &a0, v4 &a1, v4 &a2, v4 &a3 ) ALWAYS_INLINE;
 
-/* #   define BINARY(op,instr)                                            \ */
-/*     inline v4 operator op( const v4 &a, const v4 &b )                  \ */
-/*     {								       \ */
-/*       v4 c;                                                            \ */
-/*       instr;                                                           \ */
-/*       return c;                                                        \ */
-/*     } */
+    // v4int miscellaneous friends
 
-/*     BINARY(+, c.v = vec_add( a.v, b.v ) ) */
-/*     BINARY(-, c.v = vec_sub( a.v, b.v ) ) */
-/*     BINARY(*, c.v = vec_mul( a.v, b.v ) ) */
-/*       // BINARY(*, c.v = vec_madd( a.v, b.v, _zero ) ) */
+    friend inline v4    czero( const v4int &c, const v4 &a ) ALWAYS_INLINE;
+    friend inline v4 notczero( const v4int &c, const v4 &a ) ALWAYS_INLINE;
+    friend inline v4    merge( const v4int &c, const v4 &a, const v4 &b ) ALWAYS_INLINE;
 
-/* #   undef BINARY */
-    // end hacks
-    // -----------------------------------------------------------------------------
+    // v4 memory manipulation friends
 
-    // v4 miscellenous friends
+    friend inline void   load_4x1( const void * ALIGNED(16) p,
+                                   v4 &a ) ALWAYS_INLINE;
 
-    friend inline int any( const v4 &a );
-    friend inline int all( const v4 &a );
-    template<int n>
-    friend inline v4 splat( const v4 &a );
-    // friend inline v4 splat( const v4 &a, int n );
-    template<int i0, int i1, int i2, int i3>
-    friend inline v4 shuffle( const v4 &a );
-    // friend inline v4 shuffle( const v4 &a,
-    //                           int i0, int i1, int i2, int i3 );
-    friend inline void swap( v4 &a, v4 &b );
-    friend inline void transpose( v4 &a0, v4 &a1, v4 &a2, v4 &a3 );
+    friend inline void  store_4x1( const v4 &a,
+                                   void * ALIGNED(16) p ) ALWAYS_INLINE;
 
-    // v4int miscellaneous friends
+    friend inline void stream_4x1( const v4 &a,
+                                   void * ALIGNED(16) p ) ALWAYS_INLINE;
 
-    friend inline v4 czero(    const v4int &c, const v4 &a );
-    friend inline v4 notczero( const v4int &c, const v4 &a );
-    friend inline v4 merge(    const v4int &c, const v4 &a, const v4 &b );
+    friend inline void  clear_4x1( void * ALIGNED(16) dst ) ALWAYS_INLINE;
 
-    // v4 memory manipulation friends
-        
-    friend inline void load_4x1( const void * ALIGNED(16) p, v4 &a );
-    friend inline void store_4x1( const v4 &a, void * ALIGNED(16) p );
-    friend inline void stream_4x1( const v4 &a, void * ALIGNED(16) p );
```

Comment:
Am I right in thinking the changes in this first half are all a) organizational, and b) adding `ALWAYS_INLINE`?

Reply:
Yes.  I wanted to get a uniform format and style across the various v4 implementations so I could get a good and useful visual diff across different v4 intrinsics versions.  For instance, I can compare v4_altivec.h with either v4_neon.h or v4_avx2.h and see where there are holes in the completeness of the various intrinsics versions i.e. where altivec has an intrinsics implementation for some function and avx2 or neon only has a portable implementation of that function.  Basically, since I was working on neon, I took the time to do some cleanup on the other v4 intrinsics versions.